dan_layer/consensus/src/hotstuff/on_ready_to_vote_on_local_block.rs

```diff
@@ -795,16 +864,31 @@ where TConsensusSpec: ConsensusSpec
         transaction: &Transaction,
         local_committee_shard: &CommitteeShard,
     ) -> Result<(), HotStuffError> {
+        // We ignore inputs without version
+        let write_inputs: Vec<SubstateAddress> = transaction
+            .inputs()
+            .iter()
+            .chain(transaction.filled_inputs())
+            .filter(|i| i.version().is_some())
+            .map(|i| i.to_substate_address())
+            .collect();
         SubstateRecord::try_unlock_many(
             tx,
             transaction.id(),
-            local_committee_shard.filter(transaction.inputs().iter().chain(transaction.filled_inputs())),
+            local_committee_shard.filter(write_inputs.iter()),
             SubstateLockFlag::Write,
         )?;
+        // We ignore inputs without version
+        let read_inputs: Vec<SubstateAddress> = transaction
+            .input_refs()
+            .iter()
+            .filter(|i| i.version().is_some())
+            .map(|i| i.to_substate_address())
+            .collect();
```

Comment:
nit: we can save several heap allocations by just passing in the iterator directly to the filter function instead of first allocating a new vec and then passing it in. But not super critical can clean up later.